### ISSUE TITLE
New-Item -Type SymbolicLink -Target a -Path b won't work in "developer mode" without elevation

### ISSUE THREAD

**Issue by michel2bec** [PowerShell/PowerShell#5000]
Steps to reproduce
------------------
Windows 10 Pro, 1703 15063.608 
"Developer mode" activated

```powershell
PS D:\Partage\tmp> New-Item -Type File a


    Répertoire : D:\Partage\tmp


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       04/10/2017     11:42              0 a


PS D:\Partage\tmp> New-Item -Type SymbolicLink -Target a -Path b
New-Item : Cette opération nécessite un privilège d'administrateur.
Au caractère Ligne:1 : 1
+ New-Item -Type SymbolicLink -Target a -Path b
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : PermissionDenied: (D:\Partage\tmp\a:String) [New-Item], UnauthorizedAccessException
    + FullyQualifiedErrorId : NewItemSymbolicLinkElevationRequired,Microsoft.PowerShell.Commands.NewItemCommand

PS D:\Partage\tmp>
```

Well this did not work, let's try with good old CMD.exe

```powershell
PS D:\Partage\tmp> cmd
Microsoft Windows [version 10.0.15063]
(c) 2017 Microsoft Corporation. Tous droits réservés.

D:\Partage\tmp>mklink b a
Lien symbolique créé pour b <<===>> a

D:\Partage\tmp>

```
Now let's do it as Administrator, to check it's not some weird syntax issue

```powershell
PS D:\Partage\tmp> ri b
PS D:\Partage\tmp> New-Item -Type SymbolicLink -Target a -Path b


    Directory: D:\Partage\tmp


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a---l       04/10/2017     11:45              0 b


PS D:\Partage\tmp>
```

Expected behavior
-----------------

As I'm in "developper mode" I expect New-Item -Type SymbolicLink to work without elevation.

Actual behavior
---------------

I get an incorrect "elevation required" error message, incorrect because mklink works in the same context

Environment data
----------------

```powershell
PS D:\Partage\tmp> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.15063.608
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.15063.608
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1


```

**Comment by nu8** [comment]
https://github.com/PowerShell/PowerShell/issues/2845

https://github.com/PowerShell/PowerShell/issues/5000
https://github.com/PowerShell/PowerShell/pull/8534

**Comment by nu8** [comment]
This can probably be closed. After some testing, this seems to work as expected with PowerShell 6.2.0 or higher.

**Comment by vexx32** [comment]
Aye, seems like it should have been fixed by #8534. Thanks! 🙂
